rpm.py

```diff
@@ -0,0 +1,853 @@
+                dependsstr_run += (
+                    "    - "
+                    "sysroot_%s %s" % (
+                        cdt["host_subdir"], cdt["glibc_ver"]))
+            else:
+                dependsstr_run += (
+                    "  run_constrained:\n    - "
+                    "sysroot_%s ==99999999999" % cdt["host_subdir"])
+
+        dependsstr = (
+            "requirements:\n" + dependsstr_build + dependsstr_host + dependsstr_run
+        )
+
+    package_l = package.lower().replace("+", "x")
+    package_cdt_name = package_l + "-" + sn
+    license, license_family = remap_license(entry["license"])
+    d = dict(
+        {
+            "version": entry["version"]["ver"],
+            "build_number": (
+                "{{ cdt_build_number|int + 1000 }}"
+                if SINGLE_SYSROOT
+                else "{{ cdt_build_number }}"
+            ),
+            "packagename": package_cdt_name,
+            "hostmachine": cdt["host_machine"],
+            "hostsubdir": cdt["host_subdir"],
+            "depends": dependsstr,
+            "rpmurl": rpm_url,
+            "srcrpmurl": srpm_url,
+            "home": entry["home"],
+            "license": license,
+            "license_family": license_family,
+            "checksum_name": cdt["checksummer_name"],
+            "checksum": entry["checksum"],
+            "summary": '"(CDT) ' + tidy_text(entry["summary"]) + '"',
+            "description": "|\n        "
+            + "\n        ".join(tidy_text(entry["description"], 78)),  # noqa
+            # Cheeky workaround.  I use ${PREFIX},
+            # ${PWD}, ${RPM} and ${RECIPE_DIR} in
+            # BUILDSH and they get interpreted as
+            # format string tokens so bounce them
+            # back.
+            "PREFIX": "{PREFIX}",
+            "RPM": "{RPM}",
+            "PWD": "{PWD}",
+            "RECIPE_DIR": "{RECIPE_DIR}",
+            "SRC_DIR": "{SRC_DIR}",
+        }
+    )
+    odir = join(output_dir, package_cdt_name)
+    try:
+        makedirs(odir)
+    except Exception:
+        pass
+    with open(join(odir, "meta.yaml"), "w") as f:
+        f.write(RPM_META.format(**d))
+    buildsh = join(odir, "build.sh")
+    with open(buildsh, "w") as f:
+        chmod(buildsh, 0o755)
+        f.write(BUILDSH.format(**d))
+    return package
+
+
+# How do we map conda names to RPM names? The issue would be if two distros
+# name their RPMs differently we probably want to hide that away from users
+# Do I want to pass just the package name, the CDT and the arch and rely on
+# expansion to form the URL? I have been going backwards and forwards here.
+def write_conda_recipe(
+    packages,
+    distro,
+    output_dir,
+    architecture,
+    recursive,
+    override_arch,
+    dependency_add,
+    config,
+    build_number,
+):
+    cdt_name = distro
+    bits = "32" if architecture in ("armv6", "armv7a", "i686", "i386") else "64"
+    base_architectures = dict({"i686": "i386"})
```

Comment:
What's the significance of `base_architectures`? Would it make sense to add a comment above?

Reply:
This was in the upstream code, so IDK. Maybe @mingwandroid can comment?